### ISSUE TITLE
Support for native code to be main()

### ISSUE THREAD

**Issue by pauldotknopf** [pauldotknopf/net-core-qml#13]
Right now it is expected that .NET will be the entry point of the application.

Mono and .NET Core support hosting the runtime manually within native processes.

We should support a couple of headers that could be used to run a VM and configure .NET/Qml types from within main.cpp.

**Comment by RichiCoder1** [comment]
I would love this actually. Would allow us to bootstrap our native video library, but do all the interesting business in C# & Qml.

**Comment by pauldotknopf** [comment]
Yeah, this is one of the higher-priority tasks that I'd like to get done. I'll update this issue when I get to it, so stay subscribed.

**Comment by RichiCoder1** [comment]
Awesome! No pressure, I'm just amazed this project exists :). Keep up the good work and hopefully I can contribute at some point.

Have you thought about doing something like Refit and dynamically generating code rather than using reflection for interop-business? Would be cheaper and illink-able in theory.

**Comment by pauldotknopf** [comment]
> Have you thought about doing something like Refit and dynamically generating code

Yup, see #39 

Performance is totally acceptable ATM, but using ```Reflection.Emit``` would make it even better.

I've never seen/heard of Refit though, looks interesting.

**Comment by pauldotknopf** [comment]
Waiting for feedback from the .NET team to decide how to implement this.

See: https://github.com/dotnet/core-setup/issues/4530

If that proposal isn't accepted, then I will manually include the source for core-setup in this repo, which would contain all the classes/tools/etc needed to resolve deps.json and runtimeconfig.json to produce a runtime similar to what you'd get when running the ```dotnet exec``` command.

**Comment by pauldotknopf** [comment]
Check this directory for documentation/samples on how to do this.

https://github.com/pauldotknopf/Qml.Net/tree/develop/samples/hosting
